dsymbol/src/dsymbol/conversion/second.d

```diff
@@ -180,29 +180,27 @@ do
 	while (!lookup.breadcrumbs.empty)
 	{
 		auto back = lookup.breadcrumbs.back;
-		immutable bool isArr = back == ARRAY_SYMBOL_NAME;
-		immutable bool isAssoc = back == ASSOC_ARRAY_SYMBOL_NAME;
-		immutable bool isFunction = back == FUNCTION_SYMBOL_NAME;
-		if (back == POINTER_SYMBOL_NAME)
-		{
-			lastSuffix.isPointer = true;
-			lookup.breadcrumbs.popBack();
-			continue;
-		}
-		if (!isArr && !isAssoc && !isFunction)
-			break;
-		immutable qualifier = isAssoc ? SymbolQualifier.assocArray
-			: (isFunction ? SymbolQualifier.func : SymbolQualifier.array);
+		SymbolQualifier qualifier;
+		if (back == ARRAY_SYMBOL_NAME) qualifier = SymbolQualifier.array;
+		else if (back == ASSOC_ARRAY_SYMBOL_NAME) qualifier = SymbolQualifier.assocArray;
+		else if (back == FUNCTION_SYMBOL_NAME) qualifier = SymbolQualifier.func;
+		else if (back == POINTER_SYMBOL_NAME) qualifier = SymbolQualifier.pointer;
+		else break;
+
```

Comment:
Consinder using a switch case instead.

Reply:
Doean't work since the values are created at runtime

Reply:
Ah right